library/src/main/java/com/novoda/downloadmanager/LiteDownloadMigrationService.java

```diff
@@ -16,23 +16,30 @@
 
 public class LiteDownloadMigrationService extends Service {
 
+    private static final String TAG = "MigrationService";
     private Migrator v1ToV2Migrator;
     private ExecutorService executor;
```

Comment:
If you make this executor to be static and you only create one when there is no instance, then you can avoid the Migrator to be a singleton. Might we worth to double check that my assumption is correct and that if there is a task running in the executor and you kill the service, if you start the service again then the executor instance is the same...

Reply:
Good thinking! Reduce the codz 😄